### PR TITLE
Break out configuration to config file.

### PR DESCRIPTION
**Pull request by bcgraham**
This is what I've put together so far. Let me know if this along the lines of what you had in mind. 

There is a reference to `jsnell` in [game.js](https://github.com/jsnell/terra-mystica/blob/11d3ae96239f4986528fa3a6c67098eb5f0eb62f/stc/game.js#L1920) as a shortcut for "plausibly needs to see the 'Chat' button." Move that heuristic server-side? Template the js? Both seemed excessive, and it doesn't really affect anything, anyhow.

### CODE REVIEW

File: src/Server/EditGame.pm [hunk 1]
```diff
@@ -78,7 +79,7 @@ method check_user_is_not_deadbeat($dbh, $read_id, $username) {
         }
     }
 
-    my %blacklist = map { ($_ => 1) } qw(deedeebyrd);
+    my %blacklist = map { ($_ => 1) } $config{blacklist};
```
Comment (jsnell) [inline src/Server/EditGame.pm]: I think this needs to be @{$config{blacklist}}

File: src/Util/SiteConfig.pm [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+package Util::SiteConfig;
+use Exporter::Easy (EXPORT => [ '%config' ]);
+
+use JSON;
+
+use vars qw(%config);
+
+{
+  my $config_fn = 'config.json';
```
Comment (jsnell) [inline src/Util/SiteConfig.pm]: Reading the config from the current working directory seems problematic for the cronjobs. I'd prefer for this to be an absolute pathname read from an environment variable (and die-ing with an error if the environment variable is not set). The environment variable could be defaulted in app.fcgi if not set in the Apache config, something like:

TM_CONFIG=${TM_CONFIG:-$(dirname $0)/config.json}

**Comment by jsnell** [comment]
That looks great, thanks! The chat button is just a cosmetic issue, so I'm fine leaving it as-is.

The thing I can't make up my mind on is whether the config file should be in git. Seems like a very likely place for unnecessary merge conflicts. Might be better to just have a config/config-example.json file in git, and have the admin create their own config/config.json based on the example. What's your feeling on that?

**Comment by bcgraham** [comment]
I agree re: unnecessary merge conflicts; config/config-example.json seems like the way to go.

I've updated re: your feedback; let me know if that's the sort of thing you had in mind.

**Comment by jsnell** [comment]
Looks good to me, merged. Thanks!

**Pull request merged**